### PR TITLE
Backport 10440 to rec 4.5.x: Change nsec3-max-iterations default to 150 

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

Backport of #10440 

Don't forget to forward port the docs changes!

### Checklist

I have:
- [X ] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/recursordist/docs/settings.rst [hunk 1]
```diff
@@ -1356,11 +1356,15 @@ without consulting authoritative servers.
 .. versionadded:: 4.1.0
 
 -  Integer
--  Default: 2500
+-  Default: 150
 
 Maximum number of iterations allowed for an NSEC3 record.
 If an answer containing an NSEC3 record with more iterations is received, its DNSSEC validation status is treated as Insecure.
 
+.. versionchanged:: 4.5.2
```
Comment (rgacogne) [inline pdns/recursordist/docs/settings.rst:1364]: That changes will never appear in the online docs, right?

File: pdns/recursordist/docs/upgrade.rst [hunk 2]
```diff
@@ -4,9 +4,19 @@ Upgrade Guide
 Before upgrading, it is advised to read the :doc:`changelog/index`.
 When upgrading several versions, please read **all** notes applying to the upgrade.
 
-4.4.x to 4.5.0 or master
+4.5.x to 4.6.0 or master
 ------------------------
 
+4.5.1 to 4.5.2
```
Comment (rgacogne) [inline pdns/recursordist/docs/upgrade.rst:10]: Same.

File: pdns/recursordist/docs/settings.rst (hunk 1 above)
Comment (Habbie) [reply pdns/recursordist/docs/settings.rst:1364]: indeed, that's why Otto noted above "Don't forget to forward port the docs changes!"

Comment (rgacogne) [reply pdns/recursordist/docs/settings.rst:1364]: ah!

**Review by rgacogne — APPROVED** [review]

File: pdns/recursordist/docs/settings.rst (hunk 1 above)
Comment (omoerbeek) [reply pdns/recursordist/docs/settings.rst:1364]: The plan is t include that commit into the prep for 4.5.2

**Pull request merged**